src/uct/cuda/base/cuda_iface.c

```diff
@@ -15,7 +15,17 @@ ucs_status_t
 uct_cuda_base_query_devices(uct_md_h md, uct_tl_device_resource_t **tl_devices_p,
                            unsigned *num_tl_devices_p)
 {
-    return uct_single_device_resource(md, UCT_CUDA_DEV_NAME, UCT_DEVICE_TYPE_ACC,
-                                      tl_devices_p, num_tl_devices_p);
-}
+    CUdevice cuda_device;
+    ucs_status_t status;
+
+    *tl_devices_p = NULL;
+    status = uct_single_device_resource(md, UCT_CUDA_DEV_NAME, UCT_DEVICE_TYPE_ACC,
```

Comment:
add sys_device parameter to uct_single_device_resource()
all but cuda will pass UNKNOWN